pkgs/by-name/si/silverbullet/package.nix

```
@@ -0,0 +1,39 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  deno,
+}:
+stdenv.mkDerivation (
+  finalAttrs: {
+    pname = "silverbullet";
+    version = "0.7.3";
+
+    src = fetchurl {
+      url = "https://github.com/silverbulletmd/silverbullet/releases/download/${finalAttrs.version}/${finalAttrs.pname}.js";
+      hash = "sha256-IEaflx3bzA48jv+zgXU2xw1nAIJZUWz9EF7qS5KusLU=";
+    };
+
+    dontUnpack = true;
+
+    installPhase = ''
+      mkdir -p $out/{bin,lib}
+      cp $src $out/lib/silverbullet.js
+      cp ${./silverbullet} $out/bin/silverbullet
+      substituteInPlace $out/bin/silverbullet \
+        --replace "%deno%" "${deno}/bin/deno" \
+        --replace "%silverbulletjs%" "$out/lib/silverbullet.js"
+      chmod +x $out/bin/silverbullet
```

```suggestion
      runHook preInstall
      mkdir -p $out/{bin,lib}
      cp $src $out/lib/silverbullet.js
      cp ${./silverbullet} $out/bin/silverbullet
      substituteInPlace $out/bin/silverbullet \
        --replace "%deno%" "${deno}/bin/deno" \
        --replace "%silverbulletjs%" "$out/lib/silverbullet.js"
      chmod +x $out/bin/silverbullet
      runHook postInstall
```

you must run manually run phase hooks when overriding a phase